controllers/imageclusterinstall_controller_test.go

```diff
@@ -922,6 +922,65 @@ var _ = Describe("Reconcile", func() {
 
 	})
 
+	It("Set status condition in case dataImage exists and being deleted", func() {
+		bmh := bmhInState(bmh_v1alpha1.StateAvailable)
+		bmh.Spec.ExternallyProvisioned = true
+		Expect(c.Create(ctx, bmh)).To(Succeed())
+
+		clusterInstall.Spec.BareMetalHostRef = &v1alpha1.BareMetalHostReference{
+			Name:      bmh.Name,
+			Namespace: bmh.Namespace,
+		}
+		Expect(c.Create(ctx, clusterInstall)).To(Succeed())
+		Expect(c.Create(ctx, clusterDeployment)).To(Succeed())
+
+		dataImage := bmh_v1alpha1.DataImage{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:       bmh.Name,
+				Namespace:  bmh.Namespace,
+				Finalizers: []string{"test.finalizer"}, // Prevent immediate deletion
+			},
+		}
+		Expect(c.Create(ctx, &dataImage)).To(Succeed())
+
+		defer func() {
+			dataImage.SetFinalizers([]string{}) // Remove finalizer
+			Expect(c.Update(ctx, &dataImage)).To(Succeed())
+		}()
+
+		Expect(c.Delete(ctx, &dataImage)).To(Succeed())
+		Eventually(func() *metav1.Time {
+			err := c.Get(ctx, client.ObjectKey{Name: dataImage.Name, Namespace: dataImage.Namespace}, &dataImage)
+			if err != nil {
+				return nil
+			}
+			return dataImage.DeletionTimestamp
+		}).ShouldNot(BeNil()) // Ensure deletion timestamp is set
```

Comment:
I wasn't aware that this is how this worked.

I mean I guess it does, but I thought the fake client delete just removed the object rather than being aware of finalizers and such.

Reply:
I know, that's why I added this check, to make sure the fake client is aware of the finalizer and the object isn't deleted.
At first I tried to set the DeletionTimestamp but I got:
`Unable to edit test-bmh: metadata.deletionTimestamp field is immutable`
So had to come up with this...